### ISSUE TITLE
Null reference on Microsoft.Data.SqlClient.SqlClientFactory.Instance

### ISSUE THREAD

**Issue by jogibear9988** [dotnet/SqlClient#1424]
### Describe the bug

I try to register SqlClient as a Dataprovider in my NetCore6 Application, but the Field Microsoft.Data.SqlClient.SqlClientFactory.Instance is null. It has worked before, but does not any more, since I implemented my own AssemblyResolver.

Where in the Code is the instance Assigned, so I can debug whats wrong. I get no exception when Loading the DLL, or before accessing that static field (I've enabled break on all exceptions in Visual Studio)

### To reproduce
At the moment it will be hard to create a reproducible scenario. I'd firstly try to debug myself, if someone explains to me where the Field is assigned.

### Further technical details
Microsoft.Data.SqlClient version: 4.0.0
.NET target: NET6.0
SQL Server version: doesn't matter
Operating system:  Windows 10

**Comment by jogibear9988** [comment]
Okay, if I look here: 
https://github.com/dotnet/SqlClient/blob/c56baaabb641de6dd595203f78a2e93b65ac2317/src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlClientFactory.cs#L13

the Fcatory is assigned immediately, but if I open the DLL on my system in ILSpy it looks like this:

![image](https://user-images.githubusercontent.com/364896/145106363-f8c4e656-0ba7-43ed-891a-b4d86d8b836a.png)

**Comment by jogibear9988** [comment]
Ahhh....

now I found the dll in 
runtimes\win\lib\netcoreapp3.1

**Comment by jogibear9988** [comment]
so I load the wrong one.

**Comment by gumbarros** [comment]
I also having the same problem on Pop!_OS 22.10

**Comment by kvisscher** [comment]
@gumbarros I'm having the same issue on Pop 22.04. Did you manage to solve your issue?

**Comment by gumbarros** [comment]
> @gumbarros thanks for the quick reply!
> 
> I actually just found another related issue that exactly matched my issue: #1643. Hopefully it can help someone in the future.
> 
> I installed dotnet via the apt repository, but for some reason, this didn't work. I reinstalled with the install script: https://learn.microsoft.com/en-us/dotnet/core/install/linux-scripted-manual#scripted-install and everything is up and running again.

Sorry I saw the notification on my phone and confuded with this issue https://github.com/microsoft/azuredatastudio/issues/11249#issuecomment-1203329443.

To solve #1424 I installed .NET manually in my OS.

**Comment by xiaoyuvax** [comment]
> 

I have almost the same problem! now on net7.0.
I found the problem only occurs if use own AssemblyResolver!!  with the exception:

> System.InvalidOperationException:“The requested .NET Data Provider's implementation does not have an Instance field of a System.Data.Common.DbProviderFactory derived type.”

In my project, i met the same issue when System.Data.SqlClient is only loaded by my own plugin assembly loader, where the   AppDomain.CurrentDomain.AssemblyResolve delegate is used to resolve assemblies.

If i reference System.Data.SqlClien directly as nuget package in my entry assembly, it works well. but not possible as in plugin.
I think there's something wrong in underlying assembly resolving process, when use own assembly resolver.

I replaced System.Data.SqlClient with Microsoft.Data.SqlClient, it's the same issue.
